cql3/restrictions/partition_key_restrictions.cc

```diff
@@ -10,9 +10,30 @@
 
 namespace cql3 {
 namespace restrictions {
+extern logging::logger rlogger;
+
+using namespace expr;
+
+static void assert_contains_only_partition_key_columns(const expr::expression& e) {
+    const column_value* non_partition_column =
+        find_in_expression<column_value>(e, [](const column_value& cval) { return !cval.col->is_partition_key(); });
+
+    if (non_partition_column != nullptr) {
+        on_internal_error(
+            rlogger,
+            format("partition_key_restrictions were given an expression that contains a non-parition column: {}",
+                   non_partition_column->col->name_as_text()));
+    }
+}
```

Comment:
Isn't this impossible?

Reply:
it's a sanity check, same as is_prepared